src/Language/PureScript/CodeGen/JS/Common.hs

```diff
@@ -75,107 +81,160 @@ identCharToString c = '$' : show (ord c)
-
-moduleNameToJs :: ModuleName -> String
-moduleNameToJs (ModuleName pns) =
-  let name = intercalate "_" (runProperName `map` pns)
-  in if properNameIsJsReserved name then "$$" ++ name else name
+  name `elem` jsAnyReserved
 
 -- |
--- Checks whether a proper name is reserved in Javascript.
+-- Checks whether a name matches a built-in value in Javascript.
 --
-properNameIsJsReserved :: String -> Bool
-properNameIsJsReserved name =
-  name `elem` [ "Infinity"
-              , "NaN"
-              , "Object"
-              , "Function"
-              , "Boolean"
-              , "Error"
-              , "EvalError"
-              , "InternalError"
-              , "RangeError"
-              , "ReferenceError"
-              , "SyntaxError"
-              , "TypeError"
-              , "URIError"
-              , "Number"
-              , "Math"
-              , "Date"
-              , "String"
-              , "RegExp"
-              , "Array"
-              , "Int8Array"
-              , "Uint8Array"
-              , "Uint8ClampedArray"
-              , "Int16Array"
-              , "Uint16Array"
-              , "Int32Array"
-              , "Uint32Array"
-              , "Float32Array"
-              , "Float64Array"
-              , "ArrayBuffer"
-              , "DataView"
-              , "JSON"
-              , "Intl" ]
+nameIsJsBuiltIn :: String -> Bool
+nameIsJsBuiltIn name =
+  elem name
+    [ "Infinity"
+    , "NaN"
+    , "undefined"
+    , "eval"
+    , "uneval"
+    , "isFinite"
+    , "isNaN"
+    , "parseFloat"
+    , "parseInt"
+    , "decodeURI"
+    , "decodeURIComponent"
+    , "encodeURI"
+    , "encodeURIComponent"
+    , "escape"
+    , "unescape"
+    , "Object"
+    , "Function"
+    , "Boolean"
+    , "Symbol"
+    , "Error"
+    , "EvalError"
+    , "InternalError"
+    , "RangeError"
+    , "ReferenceError"
+    , "SyntaxError"
+    , "TypeError"
+    , "URIError"
+    , "Number"
+    , "Math"
+    , "Date"
+    , "String"
+    , "RegExp"
+    , "Array"
+    , "Int8Array"
+    , "Uint8Array"
+    , "Uint8ClampedArray"
+    , "Int16Array"
+    , "Uint16Array"
+    , "Int32Array"
+    , "Uint32Array"
+    , "Float32Array"
+    , "Float64Array"
+    , "Map"
+    , "Set"
+    , "WeakMap"
+    , "WeakSet"
+    , "SIMD"
+    , "ArrayBuffer"
+    , "DataView"
+    , "JSON"
+    , "Promise"
+    , "Generator"
+    , "GeneratorFunction"
+    , "Reflect"
+    , "Proxy"
+    , "Intl"
+    , "arguments"
+    ]
+
+jsAnyReserved :: [String]
+jsAnyReserved =
+  concat
+    [ jsReserved
+    , jsFutureReserved
+    , jsFutureReservedStrict
+    , jsOldReserved
+    , jsLiterals
+    ]
+
+jsReserved :: [String]
+jsReserved =
+  [ "break"
+  , "case"
+  , "catch"
+  , "class"
+  , "const"
+  , "continue"
+  , "debugger"
+  , "default"
+  , "delete"
+  , "do"
+  , "else"
+  , "export"
+  , "extends"
+  , "finally"
+  , "for"
+  , "function"
+  , "if"
+  , "import"
+  , "in"
+  , "instanceof"
+  , "let"
```

Comment:
Including `let` and `yield` in here is a bit inappropriate. I'd prefer to see `let`, `yield`, and `static` in their own list of `jsSometimesReserved`.